xdp/src/tx_loop.rs

```diff
@@ -52,8 +52,13 @@ pub fn tx_loop<T: AsRef<[u8]>>(
     let queue = dev
         .open_queue(queue_id)
         .expect("failed to open queue for AF_XDP socket");
-    let rx_size = queue.rx_size();
-    let tx_size = queue.tx_size();
+    let RingSizes {
+        rx: rx_size,
+        tx: tx_size,
+    } = queue
+        .ring_sizes()
+        // some drivers (eg bond), don't have a set ring size
+        .unwrap_or_default();
```

Comment:
might be nice to drop a log here to indicate we're using default rings

Reply:
done